src/main.rs

```diff
@@ -575,8 +575,11 @@ fn print_stats(args: &Args, values: &mut [Timing], metric_name: &str, show_perce
 }
 
 async fn process<P: Processor>(args: &Args, stopped: Arc<AtomicBool>, processor: P) -> Result<()> {
+    let batch_size = processor.get_batch_size();
+    let batch_count = args.num_vectors.div_ceil(batch_size);
+
     let multiprogress = MultiProgress::new();
-    let progress_bar = multiprogress.add(ProgressBar::new(args.num_vectors as u64));
+    let progress_bar = multiprogress.add(ProgressBar::new(batch_count as u64));
```

Comment:
```suggestion
    let progress_bar = multiprogress.add(ProgressBar::new(args.num_vectors as u64));
```

while the name `num_vectors` is not exactly fitting, we reuse this parameter for the number of unit operations to run. If I were to run 100k searches, with 16 searches per batch, I just want to use
```
bfb -n 100k --skip-setup --search --search-batch-size 16
```
And I would expect the progress bar to report in number of searches, not number of batches. i.e. #/100000, not #/6250

Reply:
The best way to do it is to decouple RPS calculation from progress bar.  Right now they are coupled.

Reply:
Do you mean we should make 100k RPC requests?  It is inconsistent with upload batching.

Reply:
Not 100k RPC requests, but 100k queries. Such that one point is the unit of insert, and one query is the unit of search